### PR TITLE
modulize server and router

### PR DESCRIPTION
**Pull request by TwinIsland**
I seperate the controller functions from the router file, i.e. `server.py` to make the code cleaner and more easy-reading

### CODE REVIEW

File: engine/server.py [hunk 1]
```diff
@@ -461,30 +74,107 @@ async def initialize_project(base_dir: str) -> ReturnDict:
...
+@app.post("/list_projects", tags=["project"])
+async def list_project() -> ReturnList:
+    """
+    list all projects
+
+    """
+    return project_utils.list_projects()
+
+
+@app.post("/remove_project_by_id", tags=["project"])
```
Comment (TwinIsland) [inline engine/server.py:108]: the previous remove project function has been changed to remove_project_by_id

File: engine/server.py [hunk 2]
```diff
@@ -461,30 +74,107 @@ async def initialize_project(base_dir: str) -> ReturnDict:
...
+    """
+    remove the project
+
+    @param task_id: if for task
+    """
+    return project_utils.remove_project_dir(task_id=task_id)
+
+
+@app.post("/remove_project", tags=["project"])
+async def remove_project(project_name: str) -> ReturnStatus:
```
Comment (TwinIsland) [inline engine/server.py:119]: this is the new remove project router, you can remove project via project name now

Comment (Tesseract22) [reply engine/server.py:119]: what is the difference betwwen the two api

Comment (TwinIsland) [reply engine/server.py:119]: user can delete the projects directly without init the project

Comment (Tesseract22) [reply engine/server.py:119]: Alright, make sense!

**Pull request merged**